### PR TITLE
Meeting Title

### PR DESCRIPTION
**Pull request by akoushke**

### CODE REVIEW

File: src/MeetingsSDKAdapter.js [hunk 1]
```diff
@@ -157,6 +160,44 @@ export default class MeetingsSDKAdapter extends MeetingsAdapter {
     }
   }
 
+  /**
+   * Fetches and returns a promise of a meeting title from a given destination
```
Comment (lalli-flores) [inline src/MeetingsSDKAdapter.js]:
```suggestion
   * Returns a promise of a meeting title for a given destination.
   * Supported destinations are person ID, room ID and SIP URI.
```

File: src/MeetingsSDKAdapter.js [hunk 2]
```diff
@@ -157,6 +160,44 @@ export default class MeetingsSDKAdapter extends MeetingsAdapter {
...
+   *
+   * @param {string} destination destination to start the meeting at
```
Comment (lalli-flores) [inline src/MeetingsSDKAdapter.js]:
```suggestion
   * @param {string} destination  Virtual meeting destination
```

File: src/MeetingsSDKAdapter.js [hunk 3]
```diff
@@ -157,6 +160,44 @@ export default class MeetingsSDKAdapter extends MeetingsAdapter {
...
+   * @returns {string}
```
Comment (lalli-flores) [inline src/MeetingsSDKAdapter.js]:
```suggestion
   * @returns {Promise<string>}
```

File: src/MeetingsSDKAdapter.js [hunk 4]
```diff
@@ -157,6 +160,44 @@ export default class MeetingsSDKAdapter extends MeetingsAdapter {
...
+   * @memberof MeetingsSDKAdapter
+   */
+  async fetchMeetingTitle(destination) {
+    const {id, type} = deconstructHydraId(destination);
+    let meetingTitle = destination;
+
+    if (type === HYDRA_ID_TYPE_PEOPLE) {
+      const {displayName} = await this.datasource.people.get(id);
+
+      meetingTitle = `${displayName}'s Personal Space`;
```
Comment (lalli-flores) [inline src/MeetingsSDKAdapter.js]:
```suggestion
      meetingTitle = `${displayName}'s Personal Room`;
```

File: src/MeetingsSDKAdapter.js [hunk 5]
```diff
@@ -157,6 +160,44 @@ export default class MeetingsSDKAdapter extends MeetingsAdapter {
...
+      meetingTitle = title;
+    } else {
+      try {
+        const people = await this.datasource.people.list({destination});
+
+        if (!!people.items && people.items.length > 0) {
+          const [person] = people.items;
+          const {displayName} = person;
+
+          meetingTitle = `${displayName}'s Personal Space`;
```
Comment (lalli-flores) [inline src/MeetingsSDKAdapter.js]:
```suggestion
          meetingTitle = `${displayName}'s Personal Room`;
```

File: src/MeetingsSDKAdapter.js [hunk 6]
```diff
@@ -157,6 +160,44 @@ export default class MeetingsSDKAdapter extends MeetingsAdapter {
...
+        }
+      } catch (error) {
+        // eslint-disable-next-line no-console
+        console.warn(`Meeting title is set to it's given destination: '${destination}'`);
```
Comment (lalli-flores) [inline src/MeetingsSDKAdapter.js]: I don't think it's necessary to log out that we didn't transform the title. We shouldn't do anything on error

File: src/MeetingsSDKAdapter.js [hunk 7]
```diff
@@ -157,6 +160,44 @@ export default class MeetingsSDKAdapter extends MeetingsAdapter {
...
+      meetingTitle = `${displayName}'s Personal Space`;
+    } else if (type === HYDRA_ID_TYPE_ROOM) {
+      const {title} = await this.datasource.rooms.get(id);
+
+      meetingTitle = title;
+    } else {
+      try {
+        const people = await this.datasource.people.list({destination});
+
+        if (!!people.items && people.items.length > 0) {
```
Comment (lalli-flores) [inline src/MeetingsSDKAdapter.js]: Couldn't this be simplified? And then just take the first person item if it exists? If it doesn't exist, we're already catching errors and title wouldn't change
```suggestion
        if (people.items !== undefined) {
```

File: src/MeetingsSDKAdapter.js [hunk 8]
```diff
@@ -157,6 +160,44 @@ export default class MeetingsSDKAdapter extends MeetingsAdapter {
...
+          const [person] = people.items;
+          const {displayName} = person;
```
Comment (lalli-flores) [inline src/MeetingsSDKAdapter.js]:
```suggestion
          const {displayName} = people.items[0];
```

Maybe it's just me but the previous line confuses me a bit, is there an object called person?

File: src/MeetingsSDKAdapter.js [hunk 9]
```diff
@@ -166,11 +207,12 @@ export default class MeetingsSDKAdapter extends MeetingsAdapter {
    */
   createMeeting(target) {
     return from(this.datasource.meetings.create(target)).pipe(
+      flatMap(({id, sipUri}) => from(this.fetchMeetingTitle(sipUri)).pipe(map((title) => ({ID: id, title})))),
```
Comment (lalli-flores) [inline src/MeetingsSDKAdapter.js]: Would it be clearer to use `destination`? or does `sipUri` also takes Hydra IDs, even though they're not technically SIP URIs?

File: src/MeetingsSDKAdapter.test.js [hunk 10]
```diff
@@ -211,6 +212,46 @@ describe('Meetings SDK Adapter', () => {
...
+    });
+
+    test('returns meeting sipUri as title', async () => {
+      const meetingTitle = 'sipUri';
+
+      expect(await meetingSDKAdapter.fetchMeetingTitle(meetingTitle)).toEqual(meetingTitle);
+    });
+
+    test('returns person displayName if the destination is a personID', async () => {
+      const meetingTitle = 'PEOPLE';
```
Comment (lalli-flores) [inline src/MeetingsSDKAdapter.test.js]: Can you rename for clarity?
```suggestion
      const meetingTitle = 'HYDRA_PEOPLE_ID';
```

File: src/MeetingsSDKAdapter.test.js [hunk 11]
```diff
@@ -211,6 +212,46 @@ describe('Meetings SDK Adapter', () => {
...
+
+      meetingSDKAdapter.datasource.people.get = jest.fn(() => Promise.resolve({displayName: 'displayName'}));
+
+      expect(await meetingSDKAdapter.fetchMeetingTitle(meetingTitle)).toEqual("displayName's Personal Space");
+    });
+
+    test("returns room's title if the destination is a personID", async () => {
+      const meetingTitle = 'ROOM';
```
Comment (lalli-flores) [inline src/MeetingsSDKAdapter.test.js]: Can you rename for clarity?
```suggestion
      const meetingTitle = 'HYDRA_ROOM_ID';
```

File: src/MeetingsSDKAdapter.test.js [hunk 12]
```diff
@@ -211,6 +212,46 @@ describe('Meetings SDK Adapter', () => {
...
+
+      meetingSDKAdapter.datasource.rooms.get = jest.fn(() => Promise.resolve({title: 'title'}));
+
+      expect(await meetingSDKAdapter.fetchMeetingTitle(meetingTitle)).toEqual('title');
+    });
+
+    test('returns person display name if the sipUri is an email address', async () => {
+      const meetingTitle = 'ID';
```
Comment (lalli-flores) [inline src/MeetingsSDKAdapter.test.js]: Can you rename for clarity?
```suggestion
      const meetingTitle = 'person@webex.com';
```

File: src/MeetingsSDKAdapter.test.js [hunk 13]
```diff
@@ -211,6 +212,46 @@ describe('Meetings SDK Adapter', () => {
...
+
+      meetingSDKAdapter.datasource.people.list = jest.fn(() =>
+        Promise.resolve({items: [{displayName: 'displayName'}]})
+      );
+
+      expect(await meetingSDKAdapter.fetchMeetingTitle(meetingTitle)).toEqual("displayName's Personal Space");
+    });
+  });
+
```
Comment (lalli-flores) [inline src/MeetingsSDKAdapter.test.js]: Can you add a test for when you don't do any transformations? e.g. `datasource.people.list` rejects or is `undefined`

Comment (akoushke) [reply src/MeetingsSDKAdapter.test.js]: it's the first test case

File: src/MeetingsSDKAdapter.js (hunk 9 above)
Comment (akoushke) [reply src/MeetingsSDKAdapter.js]: yes

**Pull request merged**